dissect/target/plugins/os/unix/_os.py

```diff
@@ -165,46 +165,55 @@ def domain(self) -> str | None:
     def os(self) -> str:
         return OperatingSystem.UNIX.value
 
-    def _parse_rh_legacy(self, path: Path) -> str | None:
-        hostname = None
-        file_contents = path.open("rt").readlines()
-        for line in file_contents:
-            if not line.startswith("HOSTNAME"):
-                continue
-            _, _, hostname = line.rstrip().partition("=")
-        return hostname
-
-    def _parse_hostname_string(self, paths: list[str] | None = None) -> dict[str, str] | None:
+    def _parse_hostname_string(self, paths: list[tuple[str, Callable[[Path], str] | None]] | None = None) -> dict[str, str] | None:
```

Comment:
Can you change this to return a `(hostname, domain)` tuple instead? It's silly that this returns a dictionary IMO.

Requires some minor changes here and in the `OpenBsdPlugin`.

Can just use this in the `__init__`:
```python
self._hostname, self._domain = self._parse_hostname_string(...)
```